src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -1445,6 +1466,63 @@ CreateShardMovePublications(MultiConnection *connection, List *shardList,
 }
 
 
+/*
+ * GetReplicationConnection opens a new replication connection to this node.
+ * This connection can be used to send replication commands, such as
+ * CREATE_REPLICATION_SLOT.
+ */
+static MultiConnection *
+GetReplicationConnection(char *nodeName, int nodePort)
+{
+	int connectionFlags = FORCE_NEW_CONNECTION;
+	connectionFlags |= REQUIRE_REPLICATION_CONNECTION_PARAM;
+
+	MultiConnection *connection = GetNodeUserDatabaseConnection(
+		connectionFlags,
+		nodeName,
+		nodePort,
+		CitusExtensionOwnerName(),
+		get_database_name(MyDatabaseId));
+	ClaimConnectionExclusively(connection);
+	return connection;
+}
+
+
+/*
+ * CreateReplicationSlot creates a replication slot with the given slot name
+ * over the given connection. The given connection should be a replication
+ * connection. This function returns the name of the snapshot that is used for
+ * this replication slot. When using this snapshot name for other transactions
+ * you need to keep the given replication connection open until you have used
+ * the snapshot name.
+ */
+static char *
+CreateReplicationSlot(MultiConnection *connection, char *slotname)
```

Comment:
do we want to unify this with CreateTemplateReplicationSlot?

Reply:
Yes, I'll do that in my follow up refactoring PR.